share/spice/cricket/cricket.js

```diff
@@ -0,0 +1,159 @@
+(function (env) {
+    "use strict";
+    var SCORECARD_ENDPOINT = 'https://query.yahooapis.com/v1/public/yql?q=select%20*%20from%20cricket.scorecard%20where%20match_id%3D{$1}&format=json&env=store%3A%2F%2F0TxIGQMQbObzvU4Apia0V0&callback=',
```

Comment:
> The problem is Yahoo rate limit the API calls and if we have too much requests then we might not be able to show live score.

I don't understand -- what does rate-limiting have to do with making the API call from the client? We can't allow an client -> 3rd party calls because that's a privacy breach. Also, it means that everyone who loads the page would make a direct request.

Proxying the API call through DDG is necessary and let's us cache the response which reduces that # of calls. If the IA ends up being popular enough to hit rate limits DDG can speak to Yahoo (they're a partner) or make the SportsRadar API a higher priority.

All the more reason to use the backend Perl approach -- DDG can cache the response for as long as needed and then we'll reduce the number of